tools/system_libs.py

```diff
@@ -333,10 +333,16 @@ def create_libcxxabi(libname):
       ['-Oz', '-I' + libcxxabi_include])
 
   # gl
-  def create_gl(libname): # libname is ignored, this is just one .o file
-    o = in_temp('gl.o')
-    check_call([shared.PYTHON, shared.EMCC, shared.path_from_root('system', 'lib', 'gl.c'), '-o', o] + get_cflags())
-    return o
+  def create_gl(libname):
+    src_dir = shared.path_from_root('system', 'lib', 'gl')
+    files = []
+    for dirpath, dirnames, filenames in os.walk(src_dir):
+      filenames = filter(lambda f: f.endswith('.c'), filenames)
+      files += map(lambda f: os.path.join(src_dir, f), filenames)
+    flags = ['-Oz']
+    if '-mt' in libname:
```

Comment:
the logic in `create_libc` is different - it checks the Setting value, then asserts on `-mt` being present or not being consistent with that.

how about creating a helper function that both of them call?

Reply:
Check